src/components/datagrid/data_grid_body.tsx

```diff
@@ -523,19 +529,41 @@ export const EuiDataGridBody: FunctionComponent<EuiDataGridBodyProps> = (
   }, [rowCount]);
 
   const wrapperRef = useRef<HTMLDivElement | null>(null);
-  const wrapperDimensions = useResizeObserver(wrapperRef.current);
+  const wrapperDimensions = useResizeObserver(wrapperRef.current, undefined);
 
   useEffect(() => {
     const boundingRect = wrapperRef.current!.getBoundingClientRect();
 
-    if (boundingRect.height !== unconstrainedHeight) {
+    if (
+      boundingRect.height !== unconstrainedHeight &&
+      height !== boundingRect.height
+    ) {
       setHeight(boundingRect.height);
     }
-    if (boundingRect.width !== unconstrainedWidth) {
+    if (
+      boundingRect.width !== unconstrainedWidth &&
+      width !== boundingRect.width
+    ) {
```

Comment:
This change introduces a dependency on `height` & `width` to the useEffect, which would need to be ignored. The previous version should function identically, as `setState` is effectively a no-op if called with its present value.

Reply:
I added this because it's a quite costly no-op as the grid is re-rendering as a result of the state setting (even if the value itself isn't changing). Another way would be to move this part https://github.com/chandlerprall/eui/pull/18/files#diff-7fd629564b805d7d5c25b7c9bc1e570afc0d443fa880192698fc13fe9776a729R588-R623 into a memoized component, but I didn't like the resulting component because a lot of props have to be passed down.

Reply:
The noop includes not triggering another render - https://codesandbox.io/s/laughing-cohen-suyii?file=/src/App.js

https://user-images.githubusercontent.com/313125/105218891-fdc4bc80-5b12-11eb-8a36-70271bcc9ccf.mov

Reply:
Hah, TIL. I was under the impression I tested this and it re-rendered, guess I confused it with some other change I had going on. My bad, I will adjust the PR.

Reply:
No worries, I had to triple check my statements :)